PhysicsTools/PatUtils/plugins/L1ECALPrefiringWeightProducer.cc

```diff
@@ -135,7 +137,8 @@ void L1ECALPrefiringWeightProducer::produce(edm::StreamID, edm::Event& iEvent, c
         continue;
       if (fabs(eta_jet) > 3.)
         continue;
-
+      if (jetMaxMuonFraction_ > 0 && jet.muonEnergyFraction() > jetMaxMuonFraction_)
```

Comment:
It looks a bit counterintuitive to me that a max muon fraction of -1 can mean "all muons," in fact.

I would have rather set such a max fraction to 1 for it (let say "999." or so in the config", so that the first part of the condition can be removed).

On the other hand, this escamotage can avoid uselessly computing the `jet.muonEnergyFraction()`, and as such it improves the performance of the code.

Still I would have preferred something like
```suggestion
      if (jetMaxMuonFraction_ < 1 && jet.muonEnergyFraction() > jetMaxMuonFraction_)
```

This being just a matter of taste, I won't consider it relevant for granting the reco signature to this PR